core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/jdbc/UniqueIDJDBCUserStoreManager.java

```diff
@@ -3472,6 +3483,312 @@ protected UniqueIDPaginatedSearchResult doGetUserListWithID(Condition condition,
+            sqlStmt = realmConfig.getUserStoreProperty(JDBCRealmConstants.GET_GROUP_FROM_GROUP_ID);
+            prepStmt = dbConnection.prepareStatement(sqlStmt);
+            prepStmt.setString(1, groupId);
+            if (sqlStmt.contains(UserCoreConstants.UM_TENANT_COLUMN)) {
+                prepStmt.setInt(2, tenantId);
+            }
+            rs = prepStmt.executeQuery();
+            if (rs.next()) {
+                groupName =  rs.getString(1);
+                createdTime = rs.getTimestamp(2, calendar).
+                        toLocalDateTime().format(DateTimeFormatter.ISO_DATE_TIME);
+                lastModified = rs.getTimestamp(3, calendar).
+                        toLocalDateTime().format(DateTimeFormatter.ISO_DATE_TIME);
+            }
+        } catch (SQLException e) {
+            String msg = "Error occurred while retrieving group id for group : " + groupName;
+            if (log.isDebugEnabled()) {
+                log.debug(msg, e);
+            }
+            throw new UserStoreException(msg, e);
+        } finally {
+            DatabaseUtil.closeAllConnections(dbConnection, rs, prepStmt);
+        }
+        if (!realmConfig.isPrimary()) {
+            groupName = StringUtils.isBlank(groupName) ? null :
+                    getMyDomainName() + UserCoreConstants.DOMAIN_SEPARATOR + groupName;
+        }
+        group.setGroupName(groupName);
+        group.setGroupID(groupId);
+        group.setCreatedDate(createdTime);
+        group.setLastModifiedDate(lastModified);
+        return group;
+    }
+
+    @Override
+    public Group doAddGroup(String groupName, String groupId, List<String> userIds, Map<String, String> claims)
+            throws UserStoreException {
+
+        persistGroup(groupName, groupId, userIds);
+        return getGroupByGroupName(groupName, null);
+    }
+
+    protected void persistGroup(String groupName, String groupId, List<String> userIds) throws UserStoreException {
+
+        Connection dbConnection = null;
+        String sqlStmt;
+
+        try {
+            dbConnection = getDBConnection();
+            sqlStmt = realmConfig.getUserStoreProperty(JDBCRealmConstants.ADD_GROUP);
+            if (sqlStmt.contains(UserCoreConstants.UM_TENANT_COLUMN)) {
+                this.updateStringValuesToDatabase(dbConnection, sqlStmt, groupId, groupName, tenantId, new Date(),
+                        new Date());
+            } else {
+                this.updateStringValuesToDatabase(dbConnection, sqlStmt, groupId, groupName, new Date(), new Date());
+            }
+
+            if (userIds != null) {
+                String[] userIdList = userIds.toArray(new String[0]);
+                // add group to user
+                String type = DatabaseCreator.getDatabaseType(dbConnection);
+                String sqlStmt2 = realmConfig.getUserStoreProperty(JDBCRealmConstants.ADD_USER_TO_ROLE_WITH_ID + "-" + type);
+
+                if (sqlStmt2 == null) {
```

Comment:
Dont we need to check for blank?

Reply:
Addressed with https://github.com/wso2/carbon-kernel/pull/3847/commits/3bb98bcb742cbac3c64ba202dbbda861a1911280